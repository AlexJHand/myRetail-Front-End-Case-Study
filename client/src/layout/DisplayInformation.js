import React from 'react';

import ImageViewer from '../components/ImageViewer';
import ProductName from '../components/ProductName';

export default class DisplayInformation extends React.Component {
    
    render() {
        return (
            <div className="layoutItem displayInformation">
                <ProductName title={this.props.title} />
                <ImageViewer images={this.props.images} title={this.props.title} />
            </div>
        )
    }
}