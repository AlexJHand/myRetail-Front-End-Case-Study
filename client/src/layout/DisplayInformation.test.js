import React from 'react';
import DisplayInformation from './DisplayInformation';
import Enzyme, {shallow, render, mount} from 'enzyme';

it("renders correctly", () => {
    const wrapper = shallow(
        <DisplayInformation 
            title="Ninja™ Professional Blender with Single Serve Blending Cups"
            images={["0"]}
        />
    )

    expect(wrapper).toMatchSnapshot();
})

it("renders child", () => {
    const wrapper = mount(
        <DisplayInformation 
            title="Ninja™ Professional Blender with Single Serve Blending Cups"
            images={
                    {
                            "AlternateImages": [{
                                    "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758_Alt01"
                                },
                                {
                                    "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758_Alt02"
                                },
                                {
                                    "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758_Alt03"
                                },
                                {
                                    "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758_Alt04"
                                },
                                {
                                    "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758_Alt05"
                                },
                                {
                                    "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758_Alt06"
                                },
                                {
                                    "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758_Alt07"
                                }
                            ],
                            "PrimaryImage": [{
                                "image": "http:\/\/target.scene7.com\/is\/image\/Target\/14263758"
                            }]
                        }
                    }
        />
        
    )

    const text = wrapper.find("h2").first().text();
    
    expect(text).toEqual("Ninja™ Professional Blender with Single Serve Blending Cups")
})