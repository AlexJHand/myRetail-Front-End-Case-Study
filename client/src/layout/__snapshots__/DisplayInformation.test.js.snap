// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="layoutItem displayInformation"
>
  <ProductName
    title="Ninja™ Professional Blender with Single Serve Blending Cups"
  />
  <ImageViewer
    images={
      Array [
        "0",
      ]
    }
    title="Ninja™ Professional Blender with Single Serve Blending Cups"
  />
</div>
`;
